src/common/Kubernetes/KubernetesClient.cs

```diff
@@ -398,8 +399,21 @@ public async Task<V1Service> CreateOrReplaceV1ServiceAsync(string namespaceName,
                 }
                 catch (HttpOperationException e) when (e.Response.StatusCode == HttpStatusCode.Conflict)
                 {
-                    await RestClient.CoreV1.DeleteNamespacedServiceAsync(service.Metadata.Name, namespaceName);
+                    try 
+                    {
+                        _log.Warning("Initial CreateNamespacedServiceAsync failed, deleting namespace");
+                        await RestClient.CoreV1.DeleteNamespacedServiceAsync(service.Metadata.Name, namespaceName);
+                    }
+                    catch (JsonException ex)

```

Comment:
@elenavillamil Please add a UT for this, KubernetesClientTests.cs exists.

Reply:
do you know how to mock DeleteNamespacedServiceAsync throwing exception? I looked at adding UT when I worked on this but got stock on that part.

Reply:
Ah figured it out. there is a throw method :)

Reply:
ug our current code uses FakeItEasy to mock things, unfortunately it seems they dont support mocking of static methods, which both  CoreV1.CreateNamespacedServiceAsync and CoreV1.DeleteNamespacedServiceAsync are. My guess this is why this code path does not have UTs to begin with. I will look at option of using other mocking tools.

[Fake It Easy Related Issue](https://github.com/FakeItEasy/FakeItEasy/issues/90)

Reply:
As a workaround, I noticed our KubernetesClient class had its own wrappers around it, so I tried changing code to use the wrappers (instead of calling IcoreV1 methods direclty) and then mock the wrappers, but when I tried to do so it i says _kClient is not mockable. From googling only interface classes can be mocked, so I tried making _kClient of type IKubernetesclient, but this causes more test to fails and doesnt solve the issue because if using interface when invoking method being tested the interface does not have the implementation for it. It becomes a chicken and egg problem (cant use object because mocking utility only support interface, but cant use interface because it doesnt have implementation for the function being tested). Hence, dont think adding UT coverage to this methods is really in scope for this PR.

Reply:
I think we can add work item for refactoring UT's or add more UT's in our backlog for now and pick it up later.